Unity.gitignore

```diff
@@ -1,36 +1,37 @@
-[Ll]ibrary/
-[Tt]emp/
-[Oo]bj/
-[Bb]uild/
-[Bb]uilds/
-Assets/AssetStoreTools*
+**/[Ll]ibrary/
+**/[Tt]emp/
+**/[Oo]bj/
+**/[Bb]uild/
+**/[Bb]uilds/
+**/Assets/AssetStoreTools*
 
-# Visual Studio cache directory
-.vs/
+# Visual Studio / Visual Studio Code cache directory
+**/.vs/
+**/.vscode/
 
 # Autogenerated VS/MD/Consulo solution and project files
-ExportedObj/
-.consulo/
-*.csproj
-*.unityproj
-*.sln
-*.suo
-*.tmp
-*.user
-*.userprefs
-*.pidb
-*.booproj
-*.svd
-*.pdb
-*.opendb
+**/ExportedObj/
+**/.consulo/
+**/*.csproj
```

Comment:
I don't believe we need to prefix `**` to expressions which just map to file extensions. Can you provide more information about this change?

Reply:
@shiftkey at the time I've tested the current .gitignore with git BASH v1.9 and v2.* and it worked fine for the ignored folders. The reason for creating a wildcard (`*/`) was that git BASH v1.9 did not seem to work with .gitignore. But it might have been the case that I did not have full admin rights.  

If you were to ask me what's the best practice for defining ignore folders and extension files for .gitignore that would be: 

Folders: 
`[Nn]ameOfIgnoredFolder/` 
`.nameOfIgnoredFolder/` 

Extnesion files: 
`*.extensionOfIgnoredFiles` 

I believe [30d40a](https://github.com/github/gitignore/pull/2850/commits/30d40a225ac122684ce8f8498d923be15e54817a) is unnecessary

Reply:
I've tested the usage of `**/` and it does not seem to make any difference. For reference I would look at [UE4 .gitignore](https://github.com/github/gitignore/blob/master/UnrealEngine.gitignore) and see how it's used there for whitelisting files